In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
import re

import pickle 
#import mglearn
import time


from nltk.tokenize import TweetTokenizer # doesn't split at apostrophes
import nltk
from nltk import Text
from nltk.tokenize import regexp_tokenize
from nltk.tokenize import word_tokenize  
from nltk.tokenize import sent_tokenize 
from nltk.corpus import stopwords
from nltk.stem import WordNetLemmatizer
from nltk.stem import PorterStemmer


from sklearn.feature_extraction.text import CountVectorizer
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.linear_model import LogisticRegression 
from sklearn.naive_bayes import MultinomialNB
from sklearn.multiclass import OneVsRestClassifier


from sklearn.model_selection import cross_val_score
from sklearn.model_selection import train_test_split
from sklearn import metrics
from sklearn.metrics import classification_report
from sklearn.metrics import confusion_matrix
from sklearn.model_selection import GridSearchCV
from sklearn.pipeline import make_pipeline

from sklearn.metrics import accuracy_score
from sklearn.svm import LinearSVC

In [2]:
train_df = pd.read_csv("wiki_movie_plots_deduped.csv", delimiter=',')
train_df.head()

Release Year                             Title Origin/Ethnicity  \
0          1901            Kansas Saloon Smashers         American   
1          1901     Love by the Light of the Moon         American   
2          1901           The Martyred Presidents         American   
3          1901  Terrible Teddy, the Grizzly King         American   
4          1902            Jack and the Beanstalk         American   

                             Director Cast    Genre  \
0                             Unknown  NaN  unknown   
1                             Unknown  NaN  unknown   
2                             Unknown  NaN  unknown   
3                             Unknown  NaN  unknown   
4  George S. Fleming, Edwin S. Porter  NaN  unknown   

                                           Wiki Page  \
0  https://en.wikipedia.org/wiki/Kansas_Saloon_Sm...   
1  https://en.wikipedia.org/wiki/Love_by_the_Ligh...   
2  https://en.wikipedia.org/wiki/The_Martyred_Pre...   
3  https://en.wikipedia.org/wiki/Terrible_Teddy,_...   
4  https://en.wikipedia.org/wiki/Jack_and_the_Bea...   

                                                Plot  
0  A bartender is working at a saloon, serving dr...  
1  The moon, painted with a smiling face hangs ov...  
2  The film, just over a minute long, is composed...  
3  Lasting just 61 seconds and consisting of two ...  
4  The earliest known adaptation of the classic f...

In [3]:
train_df['Genre length'] = train_df['Genre'].apply(len)
train_df.head()

Release Year                             Title Origin/Ethnicity  \
0          1901            Kansas Saloon Smashers         American   
1          1901     Love by the Light of the Moon         American   
2          1901           The Martyred Presidents         American   
3          1901  Terrible Teddy, the Grizzly King         American   
4          1902            Jack and the Beanstalk         American   

                             Director Cast    Genre  \
0                             Unknown  NaN  unknown   
1                             Unknown  NaN  unknown   
2                             Unknown  NaN  unknown   
3                             Unknown  NaN  unknown   
4  George S. Fleming, Edwin S. Porter  NaN  unknown   

                                           Wiki Page  \
0  https://en.wikipedia.org/wiki/Kansas_Saloon_Sm...   
1  https://en.wikipedia.org/wiki/Love_by_the_Ligh...   
2  https://en.wikipedia.org/wiki/The_Martyred_Pre...   
3  https://en.wikipedia.org/wiki/Terrible_Teddy,_...   
4  https://en.wikipedia.org/wiki/Jack_and_the_Bea...   

                                                Plot  Genre length  
0  A bartender is working at a saloon, serving dr...             7  
1  The moon, painted with a smiling face hangs ov...             7  
2  The film, just over a minute long, is composed...             7  
3  Lasting just 61 seconds and consisting of two ...             7  
4  The earliest known adaptation of the classic f...             7

In [4]:
train_df['Genre length'].describe()

count    34886.000000
mean         8.700768
std          5.453291
min          1.000000
25%          6.000000
50%          7.000000
75%          9.000000
max         81.000000
Name: Genre length, dtype: float64

In [5]:
train_df.describe()

Release Year  Genre length
count  34886.000000  34886.000000
mean    1981.314252      8.700768
std       27.815174      5.453291
min     1901.000000      1.000000
25%     1957.000000      6.000000
50%     1988.000000      7.000000
75%     2007.000000      9.000000
max     2017.000000     81.000000

In [6]:
train_df['Genre'].nunique()

2265

There are 2265 type of genres available in the given dataset

In [7]:
genre_arr = train_df['Genre'].unique()

In [8]:
genre_data = train_df['Genre'].value_counts()

In [9]:
genre_data

unknown                                                        6083
drama                                                          5964
comedy                                                         4379
horror                                                         1167
action                                                         1098
                                                               ... 
comedy, social, musical                                           1
horror, thriller, romance, musical                                1
horror, fantasy, action, magical girl, science fantasy, spy       1
biography, horror                                                 1
action, adventure, family, romance                                1
Name: Genre, Length: 2265, dtype: int64

In [10]:
type(genre_data)

pandas.core.series.Series

In [11]:
import string
alpha = string.ascii_lowercase
alpha_up = string.ascii_uppercase
print(alpha)
print(alpha_up)

abcdefghijklmnopqrstuvwxyz
ABCDEFGHIJKLMNOPQRSTUVWXYZ


In [12]:

def remove_punc(data):
    string = ''
    for i in data:
        if i in alpha:
            string = string + i
        elif i in alpha_up:
            string = string + i
        elif i == ' ':
            string = string + i
        else:
            continue
    return string

In [13]:
dat = 'In a mansion called Xanadu, part of a vast palatial estate in Florida, the elderly Charles Foster Kane is on his deathbed. Holding a snow globe, he utters a word, "Rosebud", and dies; the globe slips from his hand and smashes on the floor'
print(remove_punc(dat))

In a mansion called Xanadu part of a vast palatial estate in Florida the elderly Charles Foster Kane is on his deathbed Holding a snow globe he utters a word Rosebud and dies the globe slips from his hand and smashes on the floor


In [16]:
def genre_corr(data):
    data = data.split()
    final_gen = ''
    list_rom = ['romance','love','love story','musical b', 'romantic','rom-coms','music','musical','actionlove','romanceaction',
               'romancecomedy','romancehorror','romcom','rom\|com','rom',' \(artistic\)',"drama|romance|adult|children"
               ]
    list_act = ['act','action','adventures','kung fu','martial arts','world war ii','world war i','spy film','biker film',
               'buddy cop','buddy film','bruceploitation','drama about child soldiers',"war-time","wartime","ww1","wwii",
               'true crime','crime','\|007','gun fu','afghan war drama','actionadventure','actioncomedy','actiondrama',
                'actionlove','actionmasala','actionchildren','adventurecomedy','actionthriller','martialarts',' \(volleyball\)',
                ' \(aquatics|swimming\)',' \(aquatics|swimming\)',' \(shogi|chess\)',' (road bicycle racing)','american football',
                'dev\|nusrat jahan',' \(road bicycle racing\)','liveaction','heistcomedy','heist','historydisaster','warcomedy',
                'samurai','martial_arts','adventure','spy','superhero',"drama|romance|adult|children",'actionner',
               ]
    list_sus = ['ttriller','coming of age','coming-of-age','slice of life','psycho thriller,',"ero",'actionadventure','dramathriller',
                'dramathriller','thriler','crimethriller','actionthriller','comedysocial','erotica','erotic','comedythriller',
                  'colour\|yellow\|productions\|eros\|international',       'melodrama', 'gangsterthriller',  'ancientcostume', 
                'dramatic','biodrama','bio-drama','comedy-drama adaptation of the mordecai richler novel','drama about child soldiers',
                'drama loosely','slice of life',"comedy–drama"'actionlove','actiondrama','fantasycomedy','dramacomedy',
                'dramacomedysocial','dramathriller','comedydrama','comedyhorror','adventurecomedy','animationdrama','comedysocial',
                'erotica','erotic','biblical','biblical','colour\|yellow\|productions\|eros\|international','liveaction','superheroes',
                'heistcomedy','heist','warcomedy','dramatic','familya','familya','dramedy','dramaa','famil\|','superheroe',
                'devotionalbiography','familydrama','espionage','romancefiction','horrorthriller','suspensethriller','triller',
                'satirical','homosexual','sexual','mockumentary','periodic','politics','tv_miniseries','serial',"musical–comedy",
                "roman|porno","action—masala","horror–thriller",'family','martial_arts','horror','war','adventure','noir',
                'superhero','social','suspense',"drama|romance|adult|children",'actionner',
                ]
    list_sci = ['animated','anime','children\'s','3-d','3d','sci-fi','sci fi','science fiction','avant-garde','animationchildren',
               'computer animation',   ' in animation',   'actionchildren',  'fantasychildren\|','fantasycomedy','fantasyperiod',
                'sciencefiction','animationdrama','fantay','\|\(children\|poker\|karuta\)','superheroes','computeranimation',
                '\|\(fiction\)','science_fictionchildren','science_fiction','superhero',"drama|romance|adult|children"
                ]
    list_hor = ['psychological','j-horror','psycho thriller,',"comedy–horror",'actionadventure','comedyhorror','horror',
                ]
    for i in data:
        if i.lower() in list_rom:
            final_gen = final_gen + ' Romance'
        elif i.lower() in list_act:
            final_gen = final_gen + ' Action'
        elif i.lower() in list_sus:
            final_gen = final_gen + ' Suspense'
        elif i.lower() in list_sci:    
            final_gen = final_gen + ' Science Fiction'
        elif i.lower() in list_hor:
            final_gen = final_gen + ' Horror'
        else:
            final_gen = final_gen + ' Others'
    final_gen = set(final_gen.split())
    genre = ''
    for i in final_gen:
        genre = genre + ' {}'.format(i)
    return genre
            

In [18]:
train_df['Genre String'] = train_df['Genre'].apply(remove_punc)

In [20]:
train_df.head(20)

Release Year                                              Title  \
0           1901                             Kansas Saloon Smashers   
1           1901                      Love by the Light of the Moon   
2           1901                            The Martyred Presidents   
3           1901                   Terrible Teddy, the Grizzly King   
4           1902                             Jack and the Beanstalk   
5           1903                                Alice in Wonderland   
6           1903                            The Great Train Robbery   
7           1904                                    The Suburbanite   
8           1905                           The Little Train Robbery   
9           1905                         The Night Before Christmas   
10          1906                           Dream of a Rarebit Fiend   
11          1906  From Leadville to Aspen: A Hold-Up in the Rockies   
12          1906                                Kathleen Mavourneen   
13          1907                                       Daniel Boone   
14          1907                    How Brown Saw the Baseball Game   
15          1907                                       Laughing Gas   
16          1908                           The Adventures of Dollie   
17          1908                                    The Black Viper   
18          1908                             A Calamitous Elopement   
19          1908                               The Call of the Wild   

   Origin/Ethnicity                                  Director  \
0          American                                   Unknown   
1          American                                   Unknown   
2          American                                   Unknown   
3          American                                   Unknown   
4          American        George S. Fleming, Edwin S. Porter   
5          American                            Cecil Hepworth   
6          American                           Edwin S. Porter   
7          American                        Wallace McCutcheon   
8          American                      Edwin Stanton Porter   
9          American                      Edwin Stanton Porter   
10         American    Wallace McCutcheon and Edwin S. Porter   
11         American  Francis J. Marion and Wallace McCutcheon   
12         American                           Edwin S. Porter   
13         American   Wallace McCutcheon and Ediwin S. Porter   
14         American                                   Unknown   
15         American                      Edwin Stanton Porter   
16         American                            D. W. Griffith   
17         American                            D. W. Griffith   
18         American                             D.W. Griffith   
19         American                            D. W. Griffith   

                                 Cast                       Genre  \
0                                 NaN                     unknown   
1                                 NaN                     unknown   
2                                 NaN                     unknown   
3                                 NaN                     unknown   
4                                 NaN                     unknown   
5                           May Clark                     unknown   
6                                 NaN                     western   
7                                 NaN                      comedy   
8                                 NaN                     unknown   
9                                 NaN                     unknown   
10                                NaN                       short   
11                                NaN  short action/crime western   
12                                NaN                  short film   
13  William Craven, Florence Lawrence                biographical   
14                            Unknown                      comedy   
15    Bertha Regustus, Edward Boulden     

In [22]:
train_df['Genre Corrected'] = train_df['Genre String'].apply(genre_corr)

In [23]:
train_df.head(20)

Release Year                                              Title  \
0           1901                             Kansas Saloon Smashers   
1           1901                      Love by the Light of the Moon   
2           1901                            The Martyred Presidents   
3           1901                   Terrible Teddy, the Grizzly King   
4           1902                             Jack and the Beanstalk   
5           1903                                Alice in Wonderland   
6           1903                            The Great Train Robbery   
7           1904                                    The Suburbanite   
8           1905                           The Little Train Robbery   
9           1905                         The Night Before Christmas   
10          1906                           Dream of a Rarebit Fiend   
11          1906  From Leadville to Aspen: A Hold-Up in the Rockies   
12          1906                                Kathleen Mavourneen   
13          1907                                       Daniel Boone   
14          1907                    How Brown Saw the Baseball Game   
15          1907                                       Laughing Gas   
16          1908                           The Adventures of Dollie   
17          1908                                    The Black Viper   
18          1908                             A Calamitous Elopement   
19          1908                               The Call of the Wild   

   Origin/Ethnicity                                  Director  \
0          American                                   Unknown   
1          American                                   Unknown   
2          American                                   Unknown   
3          American                                   Unknown   
4          American        George S. Fleming, Edwin S. Porter   
5          American                            Cecil Hepworth   
6          American                           Edwin S. Porter   
7          American                        Wallace McCutcheon   
8          American                      Edwin Stanton Porter   
9          American                      Edwin Stanton Porter   
10         American    Wallace McCutcheon and Edwin S. Porter   
11         American  Francis J. Marion and Wallace McCutcheon   
12         American                           Edwin S. Porter   
13         American   Wallace McCutcheon and Ediwin S. Porter   
14         American                                   Unknown   
15         American                      Edwin Stanton Porter   
16         American                            D. W. Griffith   
17         American                            D. W. Griffith   
18         American                             D.W. Griffith   
19         American                            D. W. Griffith   

                                 Cast                       Genre  \
0                                 NaN                     unknown   
1                                 NaN                     unknown   
2                                 NaN                     unknown   
3                                 NaN                     unknown   
4                                 NaN                     unknown   
5                           May Clark                     unknown   
6                                 NaN                     western   
7                                 NaN                      comedy   
8                                 NaN                     unknown   
9                                 NaN                     unknown   
10                                NaN                       short   
11                                NaN  short action/crime western   
12                                NaN                  short film   
13  William Craven, Florence Lawrence                biographical   
14                            Unknown                      comedy   
15    Bertha Regustus, Edward Boulden     

In [24]:
train_df['Genre Corrected 2'] = train_df['Genre'].apply(genre_corr)

In [25]:
train_df.head(20)

Release Year                                              Title  \
0           1901                             Kansas Saloon Smashers   
1           1901                      Love by the Light of the Moon   
2           1901                            The Martyred Presidents   
3           1901                   Terrible Teddy, the Grizzly King   
4           1902                             Jack and the Beanstalk   
5           1903                                Alice in Wonderland   
6           1903                            The Great Train Robbery   
7           1904                                    The Suburbanite   
8           1905                           The Little Train Robbery   
9           1905                         The Night Before Christmas   
10          1906                           Dream of a Rarebit Fiend   
11          1906  From Leadville to Aspen: A Hold-Up in the Rockies   
12          1906                                Kathleen Mavourneen   
13          1907                                       Daniel Boone   
14          1907                    How Brown Saw the Baseball Game   
15          1907                                       Laughing Gas   
16          1908                           The Adventures of Dollie   
17          1908                                    The Black Viper   
18          1908                             A Calamitous Elopement   
19          1908                               The Call of the Wild   

   Origin/Ethnicity                                  Director  \
0          American                                   Unknown   
1          American                                   Unknown   
2          American                                   Unknown   
3          American                                   Unknown   
4          American        George S. Fleming, Edwin S. Porter   
5          American                            Cecil Hepworth   
6          American                           Edwin S. Porter   
7          American                        Wallace McCutcheon   
8          American                      Edwin Stanton Porter   
9          American                      Edwin Stanton Porter   
10         American    Wallace McCutcheon and Edwin S. Porter   
11         American  Francis J. Marion and Wallace McCutcheon   
12         American                           Edwin S. Porter   
13         American   Wallace McCutcheon and Ediwin S. Porter   
14         American                                   Unknown   
15         American                      Edwin Stanton Porter   
16         American                            D. W. Griffith   
17         American                            D. W. Griffith   
18         American                             D.W. Griffith   
19         American                            D. W. Griffith   

                                 Cast                       Genre  \
0                                 NaN                     unknown   
1                                 NaN                     unknown   
2                                 NaN                     unknown   
3                                 NaN                     unknown   
4                                 NaN                     unknown   
5                           May Clark                     unknown   
6                                 NaN                     western   
7                                 NaN                      comedy   
8                                 NaN                     unknown   
9                                 NaN                     unknown   
10                                NaN                       short   
11                                NaN  short action/crime western   
12                                NaN                  short film   
13  William Craven, Florence Lawrence                biographical   
14                            Unknown                      comedy   
15    Bertha Regustus, Edward Boulden     

In [26]:
train_df['Genre Corrected'].value_counts()

 Others                                    22845
 Action                                     2456
 Suspense                                   2319
 Action Others                              1744
 Others Romance                             1520
 Romance                                    1496
 Suspense Others                            1323
 Science Fiction                             323
 Science Fiction Others                      165
 Suspense Action Others                      102
 Action Romance                              100
 Action Others Romance                        92
 Suspense Others Romance                      91
 Suspense Action                              89
 Suspense Romance                             49
 Fiction Science Others                       34
                                              33
 Others Horror                                28
 Suspense Science Fiction                     18
 Suspense Science Fiction Others              13
 Suspense Action Oth

In [27]:
train_df['Genre Corrected 2'].value_counts()

 Others                                    23794
 Action                                     2314
 Suspense                                   2047
 Others Romance                             1500
 Action Others                              1455
 Romance                                    1433
 Suspense Others                            1278
 Science Fiction                             532
 Science Fiction Others                      186
 Fiction Science Others                      109
 Action Others Romance                        40
 Suspense Action Others                       33
 Others Horror                                31
                                              28
 Suspense Others Romance                      21
 Suspense Romance                             17
 Suspense Action                              16
 Action Romance                               15
 Suspense Horror                               5
 Fiction Science Action Others                 5
 Suspense Science Fi

In [28]:
train_df.drop(['Genre Corrected 2'],axis=1,inplace=True)

In [30]:
train_df.head(20)

Release Year                                              Title  \
0           1901                             Kansas Saloon Smashers   
1           1901                      Love by the Light of the Moon   
2           1901                            The Martyred Presidents   
3           1901                   Terrible Teddy, the Grizzly King   
4           1902                             Jack and the Beanstalk   
5           1903                                Alice in Wonderland   
6           1903                            The Great Train Robbery   
7           1904                                    The Suburbanite   
8           1905                           The Little Train Robbery   
9           1905                         The Night Before Christmas   
10          1906                           Dream of a Rarebit Fiend   
11          1906  From Leadville to Aspen: A Hold-Up in the Rockies   
12          1906                                Kathleen Mavourneen   
13          1907                                       Daniel Boone   
14          1907                    How Brown Saw the Baseball Game   
15          1907                                       Laughing Gas   
16          1908                           The Adventures of Dollie   
17          1908                                    The Black Viper   
18          1908                             A Calamitous Elopement   
19          1908                               The Call of the Wild   

   Origin/Ethnicity                                  Director  \
0          American                                   Unknown   
1          American                                   Unknown   
2          American                                   Unknown   
3          American                                   Unknown   
4          American        George S. Fleming, Edwin S. Porter   
5          American                            Cecil Hepworth   
6          American                           Edwin S. Porter   
7          American                        Wallace McCutcheon   
8          American                      Edwin Stanton Porter   
9          American                      Edwin Stanton Porter   
10         American    Wallace McCutcheon and Edwin S. Porter   
11         American  Francis J. Marion and Wallace McCutcheon   
12         American                           Edwin S. Porter   
13         American   Wallace McCutcheon and Ediwin S. Porter   
14         American                                   Unknown   
15         American                      Edwin Stanton Porter   
16         American                            D. W. Griffith   
17         American                            D. W. Griffith   
18         American                             D.W. Griffith   
19         American                            D. W. Griffith   

                                 Cast                       Genre  \
0                                 NaN                     unknown   
1                                 NaN                     unknown   
2                                 NaN                     unknown   
3                                 NaN                     unknown   
4                                 NaN                     unknown   
5                           May Clark                     unknown   
6                                 NaN                     western   
7                                 NaN                      comedy   
8                                 NaN                     unknown   
9                                 NaN                     unknown   
10                                NaN                       short   
11                                NaN  short action/crime western   
12                                NaN                  short film   
13  William Craven, Florence Lawrence                biographical   
14                            Unknown                      comedy   
15    Bertha Regustus, Edward Boulden     

In [31]:
train_df['GenreSplit']=train_df['Genre Corrected'].str.split()

In [33]:
train_df.head(20)

Release Year                                              Title  \
0           1901                             Kansas Saloon Smashers   
1           1901                      Love by the Light of the Moon   
2           1901                            The Martyred Presidents   
3           1901                   Terrible Teddy, the Grizzly King   
4           1902                             Jack and the Beanstalk   
5           1903                                Alice in Wonderland   
6           1903                            The Great Train Robbery   
7           1904                                    The Suburbanite   
8           1905                           The Little Train Robbery   
9           1905                         The Night Before Christmas   
10          1906                           Dream of a Rarebit Fiend   
11          1906  From Leadville to Aspen: A Hold-Up in the Rockies   
12          1906                                Kathleen Mavourneen   
13          1907                                       Daniel Boone   
14          1907                    How Brown Saw the Baseball Game   
15          1907                                       Laughing Gas   
16          1908                           The Adventures of Dollie   
17          1908                                    The Black Viper   
18          1908                             A Calamitous Elopement   
19          1908                               The Call of the Wild   

   Origin/Ethnicity                                  Director  \
0          American                                   Unknown   
1          American                                   Unknown   
2          American                                   Unknown   
3          American                                   Unknown   
4          American        George S. Fleming, Edwin S. Porter   
5          American                            Cecil Hepworth   
6          American                           Edwin S. Porter   
7          American                        Wallace McCutcheon   
8          American                      Edwin Stanton Porter   
9          American                      Edwin Stanton Porter   
10         American    Wallace McCutcheon and Edwin S. Porter   
11         American  Francis J. Marion and Wallace McCutcheon   
12         American                           Edwin S. Porter   
13         American   Wallace McCutcheon and Ediwin S. Porter   
14         American                                   Unknown   
15         American                      Edwin Stanton Porter   
16         American                            D. W. Griffith   
17         American                            D. W. Griffith   
18         American                             D.W. Griffith   
19         American                            D. W. Griffith   

                                 Cast                       Genre  \
0                                 NaN                     unknown   
1                                 NaN                     unknown   
2                                 NaN                     unknown   
3                                 NaN                     unknown   
4                                 NaN                     unknown   
5                           May Clark                     unknown   
6                                 NaN                     western   
7                                 NaN                      comedy   
8                                 NaN                     unknown   
9                                 NaN                     unknown   
10                                NaN                       short   
11                                NaN  short action/crime western   
12                                NaN                  short film   
13  William Craven, Florence Lawrence                biographical   
14                            Unknown                      comedy   
15    Bertha Regustus, Edward Boulden     

In [39]:
def text_process(word):
    """
    Takes in a string of text, then performs the following:
    1. Remove all punctuation
    2. Remove all stopwords
    3. Returns a list of the cleaned text
    """
    
    # Now just remove any stopwords
    return [i for i in word.split() if i.lower() not in stopwords.words('english')]

In [41]:
for i in range(5):
    print(train_df['Plot'][i])

A bartender is working at a saloon, serving drinks to customers. After he fills a stereotypically Irish man's bucket with beer, Carrie Nation and her followers burst inside. They assault the Irish man, pulling his hat over his eyes and then dumping the beer over his head. The group then begin wrecking the bar, smashing the fixtures, mirrors, and breaking the cash register. The bartender then sprays seltzer water in Nation's face before a group of policemen appear and order everybody to leave.[1]
The moon, painted with a smiling face hangs over a park at night. A young couple walking past a fence learn on a railing and look up. The moon smiles. They embrace, and the moon's smile gets bigger. They then sit down on a bench by a tree. The moon's view is blocked, causing him to frown. In the last scene, the man fans the woman with his hat because the moon has left the sky and is perched over her shoulder to see everything better.
The film, just over a minute long, is composed of two shots. 

In [44]:
train_df['Plot without punc'] = train_df['Plot'].apply(remove_punc)

In [50]:
train_df2 = train_df.iloc[12500:15000]
train_df2

Release Year                                    Title Origin/Ethnicity  \
12500          1995                           Miami Rhapsody         American   
12501          1995                       A Midwinter's Tale         American   
12502          1995                         Mighty Aphrodite         American   
12503          1995  Mighty Morphin Power Rangers: The Movie         American   
12504          1995                              Money Train         American   
...             ...                                      ...              ...   
14995          2006                              Mad Cowgirl         American   
14996          2006                   Madea's Family Reunion         American   
14997          2006                           Man About Town         American   
14998          2006                          Man of the Year         American   
14999          2006                         Marie Antoinette         American   

               Director                                               Cast  \
12500     David Frankel  Sarah Jessica Parker, Antonio Banderas, Mia Fa...   
12501   Kenneth Branagh                                    Michael Maloney   
12502       Woody Allen  Woody Allen, Mira Sorvino, Helena Bonham Carte...   
12503      Bryan Spicer  Karan Ashley, Johnny Yong Bosch, Steve Cardena...   
12504      Joseph Ruben  Wesley Snipes, Woody Harrelson, Jennifer Lopez...   
...                 ...                                                ...   
14995  Gregory Hatanaka           Sarah Lassez, James Duval, Walter Koenig   
14996       Tyler Perry  Tyler Perry, Blair Underwood, Lynn Whitfield, ...   
14997       Mike Binder  Ben Affleck, Rebecca Romijn, John Cleese, Jerr...   
14998    Barry Levinson                       Robin Williams, Laura Linney   
14999     Sofia Coppola       Kirsten Dunst, Jason Schwartzman, Judy Davis   

                         Genre  \
12500                   comedy   
12501                    drama   
12502                   comedy   
12503  science fiction, family   
12504                   action   
...                        ...   
14995                    drama   
14996             comedy-drama   
14997                   comedy   
14998                   comedy   
14999                biography   

                                               Wiki Page  \
12500       https://en.wikipedia.org/wiki/Miami_Rhapsody   
12501  https://en.wikipedia.org/wiki/A_Midwinter%27s_...   
12502     https://en.wikipedia.org/wiki/Mighty_Aphrodite   
12503  https://en.wikipedia.org/wiki/Mighty_Morphin_P...   
12504          https://en.wikipedia.org/wiki/Money_Train   
...                                                  ...   
14995          https://en.wikipedia.org/wiki/Mad_Cowgirl   
14996  https://en.wikipedia.org/wiki/Madea%27s_Family...   
14997  https://en.wikipedia.org/wiki/Man_About_Town_(...   
14998  https://en.wikipedia.org/wiki/Man_of_the_Year_...   
14999  https://en.wikipedia.org/wiki/Marie_Antoinette...   

                                                    Plot  Genre length  \
12500  Gwyn Marcus (Sarah Jessica Parker) has always ...             6   
12501  The story is set in Hope, Derbyshire, England....             5   
12502  The film opens on ancient Greek ruins where a ...             6   
12503  The Power Rangers participate with Bulk and Sk...            23   
12504  On Christmas, we meet foster brothers John and...             6   
...                                                  ...           ...   
14995  The central character in Mad Cowgirl is Theres...             5   
14996  After Madea (Tyler Perry) violates the terms o...            12   
14997  Top Hollywood talent agent, Jack Giamoro (Ben ...             6   
14998  Tom Dobbs is host of a satirical news program,...             6   
14999  Fourteen-year-old Maria Antonia is the beautif...             9   

                 Genre String   Genre Corrected          GenreSplit  \
12500  

In [ ]:
train_df2['Plot final clean'] = train_df2['Plot without punc'].apply(text_process)